packages/view/backend/sirius-components-view-emf/src/main/java/org/eclipse/sirius/components/view/emf/compatibility/PropertiesWidgetCreationService.java

```diff
@@ -343,6 +343,13 @@ private IStatus handleAddReferenceValues(VariableManager variableManager, Object
 
     private Object handleCreateElement(VariableManager variableManager, Object feature) {
         Optional<Object> result = Optional.empty();
+        if (this.isContainment(variableManager, feature) && !this.isMany(variableManager, feature)) {
+            // If the reference is containment and mono valued, we must first unset value before create a new one
+            EObject referenceOwner = variableManager.get(VariableManager.SELF, EObject.class).orElse(null);
+            if (referenceOwner != null && feature instanceof EReference reference) {
+                referenceOwner.eUnset(reference);
+            }
+        }
```

Comment:
I'd put this inside the `if (optionalIsChild.isPresent() && optionalEditingContext.isPresent())` block below, to avoid modifying the reference in cases where we discover a few lines later that we can not actually create the new element.